nomad/config.go

```diff
@@ -362,6 +364,36 @@ type Config struct {
 	DeploymentQueryRateLimit float64
 }
 
+func (c *Config) Copy() *Config {
+	if c == nil {
+		return nil
+	}
+
+	nc := *c
+
+	// Can't deep copy interfaces
+	// LogOutput io.Writer
+	// Logger log.InterceptLogger
+	// PluginLoader loader.PluginCatalog
+	// PluginSingletonLoader loader.PluginCatalog
+
+	nc.RPCAddr = pointer.Copy(c.RPCAddr)
```

Comment:
Note to other reviewers: we didn't bring over the `pointer` function from the original PR but it looks like it was backported previously.